events/poster.go

```diff
@@ -13,7 +14,7 @@ var (
 
 // Poster posts the event.
 type Poster interface {
-	Post(ctx context.Context, event Event)
+	Post(ctx context.Context, evt *event.Envelope)
```

Comment:
The poster shouldn't really have access to the envelope type. The envelope needs to be created by the poster.